skbio/io/registry.py

```diff
@@ -914,7 +939,8 @@ def wrapped_sniffer(
                         # Some formats may have headers which indicate their
                         # format sniffers should be able to rely on the
                         # filehandle to point at the beginning of the file.
-                        fh.seek(0)
+                        if self.header:
+                            fh.seek(0)
```

Comment:
Is this still safe for `stdin` which is not seekable?

Reply:
I think that @RaeedA added a `seekable` flag to different input types to control this behavior.

Reply:
That was the original plan, I moved some things around in the current changes to make it work more intuitively. It's now in a try/catch block.